src/dso_hdr.cc

```diff
@@ -125,6 +124,25 @@ DsoHdr::DsoHdr() {
   _file_info_vector.emplace_back(FileInfo(), 0, true);
 }
 
+namespace {
+bool string_readlink(const char *path, std::string &link_name) {
+  char buff[1024];
+  ssize_t len = ::readlink(path, buff, sizeof(buff) - 1);
```

Comment:
[Minor] You could use [std::filesystem::read_link](https://en.cppreference.com/w/cpp/filesystem/read_symlink)

Reply:
- [ ] improve readlink using cpp API